diff --git a/node_modules/studentvue/.DS_Store b/node_modules/studentvue/.DS_Store
new file mode 100644
index 0000000..d71f71f
Binary files /dev/null and b/node_modules/studentvue/.DS_Store differ
diff --git a/node_modules/studentvue/lib/.DS_Store b/node_modules/studentvue/lib/.DS_Store
new file mode 100644
index 0000000..62a2661
Binary files /dev/null and b/node_modules/studentvue/lib/.DS_Store differ
diff --git a/node_modules/studentvue/lib/StudentVue/.DS_Store b/node_modules/studentvue/lib/StudentVue/.DS_Store
new file mode 100644
index 0000000..6f712b1
Binary files /dev/null and b/node_modules/studentvue/lib/StudentVue/.DS_Store differ
diff --git a/node_modules/studentvue/lib/StudentVue/Client/Client.js b/node_modules/studentvue/lib/StudentVue/Client/Client.js
index a038a16..91be620 100644
--- a/node_modules/studentvue/lib/StudentVue/Client/Client.js
+++ b/node_modules/studentvue/lib/StudentVue/Client/Client.js
@@ -478,6 +478,7 @@
                 return {
                   courseID: course['@_CourseID']?.[0] ?? "",
                   period: Number(course['@_Period'][0]),
+                  periodRaw: course['@_Period']?.[0] ?? "",
                   title: _he.default.decode(course['@_Title'][0]),
                   room: course['@_Room'][0],
                   staff: {
@@ -614,10 +615,8 @@
           });
         };
         if (fresh || reportingPeriodIndex == null) {
-          console.log("Reporting Period: ", reportingPeriodIndex);
           mykeysbro();
           fetchBranch().then(async result => {
-            console.log("Fetched fresh data");
             const cachedData = await _asyncStorage.default.getItem("xmlCache2");
             const m = JSON.parse(cachedData ?? "{}");
             const identifier = this.district + this.username + reportingPeriodIndex + (orgYearGu ?? "");
diff --git a/node_modules/studentvue/lib/StudentVue/StudentVue.js b/node_modules/studentvue/lib/StudentVue/StudentVue.js
index 00b5314..cb38d58 100644
--- a/node_modules/studentvue/lib/StudentVue/StudentVue.js
+++ b/node_modules/studentvue/lib/StudentVue/StudentVue.js
@@ -122,7 +122,7 @@
    * @param {UserCredentials} credentials User credentials of the student
    * @returns {Promise<Client>} Returns the client and the information of the student upon successful login
    */
-  function login(districtUrl, credentials, proxyUrl = "https://studentvuelib.up.railway.app") {
+  function login(districtUrl, credentials, proxyUrl = "https://gradedurianproxytest-production.up.railway.app") {
     return new Promise((res, rej) => {
       if (districtUrl.length === 0) {
         return rej(new _RequestException.default({
diff --git a/node_modules/studentvue/lib/utils/soap/Client/Client.js b/node_modules/studentvue/lib/utils/soap/Client/Client.js
index 3458edc..43fa095 100644
--- a/node_modules/studentvue/lib/utils/soap/Client/Client.js
+++ b/node_modules/studentvue/lib/utils/soap/Client/Client.js
@@ -40,7 +40,7 @@
         encrypted: this.encrypted
       };
     }
-    constructor(credentials, Purl = "https://studentvuelib.up.railway.app") {
+    constructor(credentials, Purl = "https://gradedurianproxytest-production.up.railway.app") {
       this.__username__ = credentials.username;
       this.__password__ = credentials.password;
       this.__district__ = credentials.districtUrl;
@@ -138,7 +138,7 @@
           }
           return response.text();
         }).then(data => {
-          const parser = new _fastXmlParser.XMLParser({});
+          const parser = new _fastXmlParser.XMLParser({ processEntities: { enabled: true, maxTotalExpansions: 10000 } });
           const result = parser.parse(data);
           const parserTwo = new _fastXmlParser.XMLParser({
             ignoreAttributes: false,
@@ -147,7 +147,8 @@
             },
             processEntities: false,
             parseAttributeValue: false,
-            parseTagValue: false
+            parseTagValue: false,
+            maxEntityCount: 10000
           });
           const obj = parserTwo.parse(preparse(result['soap:Envelope']['soap:Body'].ProcessWebServiceRequestResponse.ProcessWebServiceRequestResult));
           if (defaultOptions.validateErrors && typeof obj === 'object' && 'RT_ERROR' in obj) {
@@ -215,10 +216,11 @@
           }
           return response.text();
         }).then(data => {
-          const parser = new _fastXmlParser.XMLParser({});
+          const parser = new _fastXmlParser.XMLParser({ processEntities: { enabled: true, maxTotalExpansions: 10000 } });
           const result = parser.parse(data);
           const parserTwo = new _fastXmlParser.XMLParser({
-            ignoreAttributes: false
+            ignoreAttributes: false,
+            maxEntityCount: 10000
           });
           const obj = parserTwo.parse(preparse(result['soap:Envelope']['soap:Body'].ProcessWebServiceRequestResponse.ProcessWebServiceRequestResult));
           if (defaultOptions.validateErrors && typeof obj === 'object' && 'RT_ERROR' in obj) {
@@ -281,7 +283,7 @@
           } else {
             var data = realResponse.response;
           }
-          const parser = new _fastXmlParser.XMLParser({});
+          const parser = new _fastXmlParser.XMLParser({ processEntities: { enabled: true, maxTotalExpansions: 10000 } });
           const result = parser.parse(data);
           const parserTwo = new _fastXmlParser.XMLParser({
             ignoreAttributes: false,
@@ -290,7 +292,8 @@
             },
             processEntities: false,
             parseAttributeValue: false,
-            parseTagValue: false
+            parseTagValue: false,
+            maxEntityCount: 10000
           });
           const obj = parserTwo.parse(preparse(result['soap:Envelope']['soap:Body'].ProcessWebServiceRequestMultiWebResponse.ProcessWebServiceRequestMultiWebResult));
           if (defaultOptions.validateErrors && typeof obj === 'object' && 'RT_ERROR' in obj) {
